src/services/projectService.ts

```diff
@@ -44,54 +45,48 @@ export default class ProjectService implements IProjectService {
      * Save a project
      * @param project - Project to save
      */
-    public save(project: IProject, securityToken: ISecurityToken): Promise<IProject> {
+    public async save(project: IProject, securityToken: ISecurityToken): Promise<IProject> {
         Guard.null(project);
 
-        return new Promise<IProject>(async (resolve, reject) => {
-            try {
-                if (!project.id) {
-                    project.id = shortid.generate();
-                }
-
-                const storageProvider = StorageProviderFactory.createFromConnection(project.targetConnection);
-                await this.saveExportSettings(project);
-                project = encryptProject(project, securityToken);
-
-                await storageProvider.writeText(
-                    `${project.name}${constants.projectFileExtension}`,
-                    JSON.stringify(project, null, 4),
-                );
-
-                resolve(project);
-            } catch (err) {
-                reject(err);
-            }
-        });
+        if (!project.id) {
+            project.id = shortid.generate();
+        }
+
+        const storageProvider = StorageProviderFactory.createFromConnection(project.targetConnection);
+        await this.saveExportSettings(project);
+        project = encryptProject(project, securityToken);
+
+        await storageProvider.writeText(
+            `${project.name}${constants.projectFileExtension}`,
+            JSON.stringify(project, null, 4),
+        );
+
+        return project;
```

Comment:
why get rid of try/catch block?

Reply:
The try/catch was only there to facilitate rejecting the promise in an error scenario, otherwise the exception would bubble by default.  No reason to use Promises when we can use async/await.